androidApp/src/main/java/com/mbta/tid/mbta_app/android/pages/EditFavoritesPage.kt

```diff
@@ -0,0 +1,265 @@
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.heading
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.dp
+import com.mbta.tid.mbta_app.android.R
+import com.mbta.tid.mbta_app.android.component.DirectionLabel
+import com.mbta.tid.mbta_app.android.component.HaloSeparator
+import com.mbta.tid.mbta_app.android.component.PillDecoration
+import com.mbta.tid.mbta_app.android.component.RoutePill
+import com.mbta.tid.mbta_app.android.component.RoutePillType
+import com.mbta.tid.mbta_app.android.component.routeCard.LoadingRouteCard
+import com.mbta.tid.mbta_app.android.component.routeCard.RouteCardContainer
+import com.mbta.tid.mbta_app.android.favorites.FavoritesViewModel
+import com.mbta.tid.mbta_app.android.favorites.NoFavoritesView
+import com.mbta.tid.mbta_app.android.util.IsLoadingSheetContents
+import com.mbta.tid.mbta_app.android.util.SettingsCache
+import com.mbta.tid.mbta_app.android.util.Typography
+import com.mbta.tid.mbta_app.android.util.modifiers.placeholderIfLoading
+import com.mbta.tid.mbta_app.model.LeafFormat
+import com.mbta.tid.mbta_app.model.RouteCardData
+import com.mbta.tid.mbta_app.model.RouteStopDirection
+import com.mbta.tid.mbta_app.model.response.GlobalResponse
+import com.mbta.tid.mbta_app.repositories.Settings
+import kotlinx.datetime.Clock
+
+@Composable
+fun EditFavoritesPage(
+    global: GlobalResponse?,
+    favoritesViewModel: FavoritesViewModel,
+    onClose: () -> Unit,
+) {
+    val showStationAccessibility = SettingsCache.get(Settings.StationAccessibility)
+    LaunchedEffect(Unit) { favoritesViewModel.loadStaticRouteCardData(global) }
+
+    val favoritesState by remember {
+        mutableStateOf(favoritesViewModel.favorites?.associate { it to true }?.toMutableMap())
```

Comment:
I think if you use `mutableStateMapOf` then changes within the map will actually cause components to rerender, which would let you use `AnimatedVisibility` to hide the favorites that are being deleted without having to refresh the entire ViewModel.